### DISCUSSION TITLE
Using Hyper?

### DISCUSSION THREAD

**Discussion by letsandeepio** [johnlindquist/kit#309] [Q&A] [answered]
Hi John & everyone, first of all, kudos to John for making such a brilliant & efficient piece of software. I have been using it for some time now, basically, I had some personal scripts all over the place and now I am trying to convert them into JS to use with Kit.

I see that Kit has a nice helper function for 'iTerm' which can directly help to execute commands iterm(`cd ~/projects/scriptkit.com; vercel dev`)

I wondered how that was implemented as I was looking for similar functionality but Hyper Terminal.

Thank you :D

**Comment by johnlindquist** [comment, marked as answer]
Thanks for the kind words!

Could you try this `hyper` function (just copy/paste it into a script) with whatever command you're looking to run?

```js
let hyper = async command => {
  command = `"${command.replace(/"/g, '\\"')}"`
  return await applescript(String.raw`
    if (text 1 thru 2 of${command}) is equal to "cd" then
    do shell script "open -a Hyper " & text 4 thru -1 of ${command}
    else
    do shell script "open -a Hyper ~"
    set appOpen to false
    set nbrOfTry to 0
    delay 0.5
    repeat
        try
            tell application "System Events"
                if exists (window 1 of process "Hyper") then
                    set appOpen to true
                    exit repeat
                end if
            end tell
        end try
        set nbrOfTry to nbrOfTry + 1
        if nbrOfTry = 20 then exit repeat
        delay 0.5
    end repeat
    if appOpen then tell application "System Events" to keystroke ${command} & return
    end if
    `)
}

await hyper(`echo "Pizza"`)

```

**Comment by letsandeepio** [comment]
This works great, thanks @johnlindquist! PS: Had to add a delay of 1 second before sending the keystroke (right below the end repeat and before the if statement). I have another plugin (auto-open CWD with a new tab) that was interfering.

**Reply by johnlindquist** [reply]
Just FYI, I won't be adding this to Script Kit core. It uses a "Keystroke return" approach and I'm  just not comfortable having a util which presses the "return" key in the core.

**Reply by letsandeepio** [reply]
Ah! makes total sense, I prefer it that way too: do you think I should take out the `& return` right above the end `if block`?

**Reply by johnlindquist** [reply]
You just have to weigh "microscopic chance of return sent to a modal" vs manually hitting return every time. 🤷‍♂️
